include/agent.h

```diff
@@ -0,0 +1,73 @@
+#ifndef _ABM_AGENT_H_
+#define _ABM_AGENT_H_
+
+#include <memory>
+#include <vector>
+
+#include "agent.h"
+#include "config.h"
+#include "graph.h"
+
+namespace abm {
+
+//! Agent class that has id, origin, destination, departure time, current node
+//! and compute path from current node to destination.
+class Agent {
+ public:
+  //! Construct an agent class from either an id, or id, origin and destination
+  //! \param[in] id Agent id
+  //! \param[in] origin Agent origin
+  //! \param[in] destination Agent destination
+  explicit Agent(const graph::vertex_t& id) : id_{id} {};
+  explicit Agent(const graph::vertex_t& id, const graph::vertex_t& origin,
+                 const graph::vertex_t& destination)
+      : id_{id}, origin_{origin}, destination_{destination} {};
+
+  //! Get and set id
+  graph::vertex_t id() const { return id_; }
+
+  //! Get and set origin
+  graph::vertex_t origin() const { return origin_; }
+
+  //! Get and set destination
+  graph::vertex_t destination() const { return destination_; }
+
+  //! Get and set departure time
```

Comment:
Add individual comments
```suggestion
  //! Return departure time
```